### PR TITLE
The functions now return string values.

### PR DESCRIPTION
**Pull request by vharsh**
- Improves code style
- Adds .gitignore file, to ignore images and pyc files
- Change the functions behaviour to return(used to print earlier)

### CODE REVIEW

**Comment by vharsh** [comment]
Pull request #1

File: pymdb.py [hunk 1]
```diff
@@ -5,93 +5,85 @@
...
+    """Pulls out movie names from the Imdb Top 250 webpage and adds them to a
+    dictionary with its position.
     Returns dictionary"""
 
-    movielist=dict()
-    i=1
-    url='http://www.imdb.com/chart/top'
-    file= urllib.urlopen(url).read()
-    link = re.findall('title=".*?dir.*?>(.*?)</a>',file)
+    movielist = dict()
```
Comment (the-zebulan) [inline pymdb.py:12]: `movielist = {}` (~6 times faster)

---

https://doughellmann.com/blog/2012/11/12/the-performance-impact-of-using-dict-instead-of-in-cpython-2-7-2/

> tl;dr
> 
> With CPython 2.7, using dict() to create dictionaries takes up to
> 6 times longer and involves more memory allocation operations than the
> literal syntax. Use {} to create dictionaries, especially if you
> are pre-populating them, unless the literal syntax does not work for
> your case.

File: pymdb.py [hunk 2]
```diff
@@ -5,93 +5,85 @@
...
+    file = urllib.urlopen(url).read()
+    link = re.findall('title=".*?dir.*?>(.*?)</a>', file)
     for links in link:
-        
-        movielist[i]=links
-        i+=1
+
+        movielist[i] = links
+        i += 1
     return movielist
```
Comment (the-zebulan) [inline pymdb.py:21]: `return {i: link for i, link in enumerate(links, 1)}`

File: pymdb.py [hunk 3]
```diff
@@ -5,93 +5,85 @@
...
-    url='http://www.imdb.com/chart/top'
-    file= urllib.urlopen(url).read()
+    i = 1
+    yearcount = dict()
 
-            
-         
-    years = re.findall('secondaryInfo">\((.*?)\)</span>',file)
+    url = 'http://www.imdb.com/chart/top'
+    file = urllib.urlopen(url).read()
```
Comment (the-zebulan) [inline pymdb.py:33]: This warning comes up using PyCharm: `Shadows built-in name 'file'`

This variable name should be changed.

File: pymdb.py [hunk 4]
```diff
@@ -5,93 +5,85 @@
...
-    links = re.findall('<div class=".*?tconst="(.*?)"></div>',file)
-
+    movielist = []
+    url2 = 'http://www.imdb.com/chart/top'
+    file = urllib.urlopen(url2).read()
+    links = re.findall('<div class=".*?tconst="(.*?)"></div>', file)
 
     for link in links:
         movielist.append(link)
     return movielist
```
Comment (the-zebulan) [inline pymdb.py:69]: This `for` loop doesn't seem to make much sense. `re.findall` already returns a list of links. Why iterate over it just to put each item in a list called `movielist`? Isn't `links` always going to be equal to `movielist`? You could just return `re.findall(...)`.

File: pymdb.py [hunk 5]
```diff
@@ -116,9 +108,9 @@ def getposter(self):
         To check current directort type 'os.getcwd()'
```
Comment (the-zebulan) [inline pymdb.py:108]: `directort -> directory`

File: pymdb.py [hunk 6]
```diff
@@ -129,17 +121,17 @@ def year(self):
     def rating(self):
         """Prints IMDb and RT ratings"""
         return float(self.stuff.find('movie').get("imdbRating"))
-        
+
     def RTrating(self):
```
Comment (the-zebulan) [inline pymdb.py:125]: Function name doesn't conform to PEP8 standards (`snake_case`).

`RTrating -> rt_rating`

File: pymdb.py [hunk 7]
```diff
@@ -5,93 +5,85 @@
...
-            
-         
-    years = re.findall('secondaryInfo">\((.*?)\)</span>',file)
+    years = re.findall('secondaryInfo">\((.*?)\)</span>', file)
     for year in years:
-        year250[i]=year
-        i+=1
+        year250[i] = year
+        i += 1
     return year250
```
Comment (the-zebulan) [inline pymdb.py:56]: `return {i: year for i, year in enumerate(years, 1)}`

File: pymdb.py [hunk 8]
```diff
@@ -148,20 +140,22 @@ def plot(self):
...
 
 class MovieId(Movie):
     """Takes IMDb ID as parameter instead of title"""
-    def __init__(self,id):
-        url=urllib.urlopen("http://www.omdbapi.com/?i=%s&y=&plot=short&r=xml"% id)
-        input=url.read()
+
+    def __init__(self, id):
+        url = urllib.urlopen("http://www.omdbapi.com/?i=%s&y=&plot=short&r=xml"% id)
+        input = url.read()
```
Comment (the-zebulan) [inline pymdb.py:157]: This warning comes up using PyCharm: `Shadows built-in name 'input'`

This variable name should be changed.

**Review by the-zebulan — COMMENTED** [review]
There were some good changes made by @vharsh . I just saw a few more and thought they could be fixed easily as well.

Also, I'm not sure using regular expressions is the best way to parse HTML. Maybe something like BeautifulSoup4 would be better.

**Comment by vipin14119** [comment]
I can work with BeautifulSoup4 , i had written few scripts just like that is past using bs4 , so should i change it to bs4 implementation ?
if yes then just open a issue of hackoctoberfest

**Comment by the-zebulan** [comment]
It's up to you, I was just making a few suggestions. I like using
beautifulsoup4 and requests when I've done similar scripts.

File: pymdb.py (hunk 4 above)
Comment (kaushiksk) [reply pymdb.py:69]: I know I just realized. I think back then for some reason I thought re.findall returned an iterable. Will change this.

File: pymdb.py (hunk 3 above)
Comment (kaushiksk) [reply pymdb.py:33]: Will change this.

File: pymdb.py (hunk 1 above)
Comment (kaushiksk) [reply pymdb.py:12]: I didn't know this one! Thanks !

**Comment by kaushiksk** [comment]
@vharsh  Sorry it took so long. Thanks for all the changes.

File: pymdb.py (hunk 7 above)
Comment (kaushiksk) [reply pymdb.py:56]: Thanks for pointing all this out. This was back when i was still a python noob. Never came back to it.

**Comment by kaushiksk** [comment]
@vipin14119 @the-zebulan I had written this when I was learning regex. I've been using bs4 ever since too. This serves as my personal reference for regex too. But you're free to rewrite it in bs4.
